File: Python/String/Palindrome Permutation.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        table={}
        for c in s:
            table[c]=table[c]+1 if c in table else 1
        count=0
        for k,v in table.items():
            if v%2:
                count+=1
            if count>1:
                return False
        return True

File: Java/Palindrome Permutation.java
//brute force
public class Solution {
    public boolean canPermutePalindrome(String s) {
        int count = 0;
        for (char i = 0; i < 128 && count <= 1; i++) {
            int ct = 0;
            for (int j = 0; j < s.length(); j++) {
                if (s.charAt(j) == i)
                    ct++;
            }
            count += ct % 2;
        }
        return count <= 1;
    }
}

//hashmap
public class Solution {
  public boolean canPermutePalindrome(String s) {
     HashMap < Character, Integer > map = new HashMap < > ();
     for (int i = 0; i < s.length(); i++) {
         map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0) + 1);
     }
     int count = 0;
     for (char key: map.keySet()) {
         count += map.get(key) % 2;
     }
     return count <= 1;
 }
}









